listenbrainz/db/feedback.py

```diff
@@ -0,0 +1,152 @@
+        Returns:
+            A list of user's loved and hated recordings
+    """
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+          ORDER BY created DESC
+            """), {
+                'user_id': user_id
+            }
+        )
+        while True:
+            row = result.fetchone()
+            if not row:
+                break
+
+            records.append(_format_recording_feedback_row(row[0], row[1], row[2]))
+
+    return records
+
+
+def get_feedback_single_type(user_id, score):
+    """ Get a list of recording feedbacks of particular type (loved or hated) given by the user.
+        If score is +1 then get loved recordings, if -1 then get hated recordings.
+        Args:
+            user_id (int): the row ID of the user
+        Returns:
+            A list of user's loved or hated recordings
+    """
+
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+             WHERE user_id = :user_id
+               AND score = :score
+          ORDER BY created DESC
+            """), {
+                'user_id': user_id,
+                'score': score
+            }
+        )
+        while True:
+            row = result.fetchone()
+            if not row:
+                break
+
+            records.append(_format_recording_feedback_row(row[0], row[1], row[2]))
+
+    return records
+
+
+def get_feedback_for_recording(recording_msid):
+    """ Get a list of feedbacks for a given recording.
+        Args:
+            recording_msid (string): the MessyBrainz ID of the recording
+        Returns:
+            A list of feedbacks for the given recording
+    """
+
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+             WHERE recording_msid = :recording_msid
+          ORDER BY created DESC
+            """), {
+                'recording_msid': recording_msid
+            }
+        )
+        while True:
+            row = result.fetchone()
+            if not row:
+                break
+
+            records.append(_format_recording_feedback_row(row[0], row[1], row[2]))
+
+    return records
+
+
+def _format_recording_feedback_row(user_id, recording_msid, score):
+    """ Format the given entires into a dict.
+        Args:
+            user_id (int): the row id of the user
+            recording_msid (string): the MessyBrainz ID of the recording
+            score (int): The score associated with the recording
+        Returns:
+            A dict formed from the args.
+    """
+    record = {
+        'user_id': user_id,
```

Comment:
I am somewhat inclined to return the `username` rather than the `id`. @mayhem Please let me know your opinion.

Reply:
It depends on what we want to use this data for. If it's to give to an external endpoint for a user to use, it should definitely be a musicbrainz login name. If we need to use this information for internal stuff, it might be better to be an id, so it definitely depends on the final requirement.

Reply:
why not both?

Reply:
I don't think we should return the user_id in the case of API endpoints. I guess the class approach can get it solved with the class approach. We can have to methods
1. to_api()
2. to_internal_methods() ("yet to think its name")

Reply:
right, but this is the database function, just return both here and take the one you want for the API.

Reply:
@paramsingh DB doesn't store user_name though. Makes sense to return only user_id and change it in feedback_api class, when needed?

Reply:
if you're gonna need the username later, joining in this query and returning here might be a better idea than having to run a different query later. I don't know how this will be used exactly, so I can't say.

Reply:
Standardized the returned object.